overlord/hardwarestate/hardwaremgr.go

```diff
@@ -0,0 +1,64 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package hardwarestate
+
+import (
+	"sync"
+
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+type HardwareManager struct {
+	state   *state.State
+	udevMon udevMonitor
+	once    sync.Once
+}
+
+// Manager returns a new HardwareManager.
+// Extra interfaces can be provided for testing.
+func Manager(s *state.State) (*HardwareManager, error) {
+	m := &HardwareManager{
+		udevMon: NewUDevMonitor(),
+		state:   s,
+	}
+	return m, nil
+}
+
+func (m *HardwareManager) KnownTaskKinds() []string {
+	return nil
+}
+
+// Ensure implements StateManager.Ensure.
+func (m *HardwareManager) Ensure() (err error) {
```

Comment:
Alternative idea: have the goroutine starting be done *explicitly* in daemon, this should also nullify your issues with testing. Essentially all we would do here is to make a channel for sending events.

Reply:
if it needs that extra handling then it doesn't really fit as a Manager, it's a layering violation if overlord needs to be sidestepped that way

Reply:
the issue is about Overlord vs daemon,  overlord is the place that needs to know how to glue managers, not daemon. If we need to do special things we should do them in overlord.

Reply:
Ack. We need to think how to do an overlord manager (or some other concept perhaps) that is a perpetually running IO wait. Ideas apart from hot-plug: live server-driven refresh or other event subscription.